### ISSUE TITLE
Some opencv dependencies are being overriden by system packages

### ISSUE THREAD

**Issue by mwilliammyers** [LLNL/spack#1247]
Despite us using `depends_on('foo')` in the `package.py` and `CMAKE_PREFIX_PATH` being properly set many of the deps are not being found by cmake:

```
...
GUI:
--     QT:                          NO
--     GTK+ 2.x:                    YES (ver 2.24.23)
--     GThread :                    YES (ver 2.28.8)
--     GtkGlExt:                    NO
--     OpenGL support:              NO
--     VTK support:                 NO
--
--   Media I/O:
--     ZLib:                        /usr/lib64/libz.so (ver 1.2.8)
--     JPEG:                        /usr/lib64/libjpeg.so (ver )
--     WEBP:                        build (ver 0.3.1)
--     PNG:                         /usr/lib64/libpng.so (ver 1.6.16)
--     TIFF:                        /usr/lib64/libtiff.so (ver 42 - 4.0.3)
--     JPEG 2000:                   /usr/lib64/libjasper.so (ver 1.900.1)
--     OpenEXR:                     build (ver 1.7.1)
--     GDAL:                        NO
...
```

**Comment by mwilliammyers** [comment]
At least for libpng, it can't find the headers in libpng's spack prefix so that might be why it uses the system's version...

**Comment by adamjstewart** [comment]
Does CMake have options to specify the path of each prefix? Try running `cmake -LAH` in the root directory.

**Comment by davydden** [comment]
see similar issue i had a while ago with  [hashdist system packages](https://github.com/hashdist/hashstack/issues/907#issuecomment-172484357).  That's why i was suggesting this https://github.com/LLNL/spack/issues/1224.

Note that  CMake always search `CMAKE_PREFIX_PATH` first and then any other paths prescribed as HINTS (see find [library()](https://cmake.org/cmake/help/v3.0/command/find_library.html)).

> CMAKE_PREFIX_PATH being properly set 

so does it have a path to system's package somewhere inside?

**Comment by mwilliammyers** [comment]
Fixed in #1234.
